as/NFT/assembly/index.ts

```diff
@@ -47,8 +47,16 @@ export function claim_media(tokenMetadata: TokenMetadata): Media {
     assert_deposit_attached(DESIGN_PRICE)
 
     /** Assert uniqueId is actually unique */
-
-    let design = new Media(tokenMetadata.media, tokenMetadata.extra)
+    let design = new Media(
+        tokenMetadata.title,
+        tokenMetadata.copies,
+        tokenMetadata.media,
+        tokenMetadata.extra,
+        tokenMetadata.description,
+        tokenMetadata.media_hash,
+        tokenMetadata.reference,
+        tokenMetadata.reference_hash,
+    )
```

Comment:
We have to add something else rather than reference:
```
  reference: string|null, // URL to an off-chain JSON file with more info.
```

Reference serves another scope and we would break the standard if we use it as link to the deployed crative code.

I would add a field here that servers that scope and call it something like `media_animation`